### PR TITLE
plat-stm32mp1: secure remoteproc dependancies on RCC-MCKPROT

### PR DESCRIPTION
**Pull request by etienne-lms**
Refine configuration consistency of remoteproc firmware isolation on stm32mp15:
- stm32_etzpc firewall driver ensure MCKPROT isolation consistency vs RCC[MCKPROT],
- stm32_remoteproc driver ensures RCC[MCKPROT] is enabled when needed,
- remove useless verification of parent clock secure hardening since the remoteproc driver ensures MCKPROT is enabled.

### CODE REVIEW

File: core/drivers/firewall/stm32_etzpc.c [hunk 1]
```diff
@@ -153,6 +156,43 @@ static enum etzpc_decprot_attributes etzpc_binding2decprot(uint32_t mode)
 	}
 }
 
+static void
+sanitize_decprot_config(uint32_t decprot_id __maybe_unused,
+			enum etzpc_decprot_attributes attr __maybe_unused)
+{
+#ifdef CFG_STM32MP15
+	/*
+	 * stm32MP15: check dependency on RCC TZEN/MCKPROT configuration
```
Comment (GseoC) [inline core/drivers/firewall/stm32_etzpc.c:165]: s/stm32MP15/STM32MP15

File: core/drivers/firewall/stm32_etzpc.c [hunk 2]
```diff
@@ -153,6 +156,43 @@ static enum etzpc_decprot_attributes etzpc_binding2decprot(uint32_t mode)
...
+	 * when a ETZPC resource is assignted to secure or isolated Cortex-M
```
Comment (GseoC) [inline core/drivers/firewall/stm32_etzpc.c:166]: when an ETZPC resource is secured or isolated for Cortex-M

File: core/drivers/firewall/stm32_etzpc.c [hunk 3]
```diff
@@ -153,6 +156,43 @@ static enum etzpc_decprot_attributes etzpc_binding2decprot(uint32_t mode)
...
+	case DECPROT_NS_R_S_W:
+		if (!stm32_rcc_is_secure()) {
+			IMSG("WARNING: RCC not secure, insecure ETZPC hardening %"PRIu32":%u",
+			     decprot_id, attr);
+			if (!IS_ENABLED(CFG_INSECURE))
+				panic();
+		}
+		break;
+	case DECPROT_MCU_ISOLATION:
+		if (!stm32_rcc_is_secure() || !stm32_rcc_is_mckprot()) {
```
Comment (GseoC) [inline core/drivers/firewall/stm32_etzpc.c:180]: This trace does not inform if rcc isn't secure

File: core/drivers/remoteproc/stm32_remoteproc.c [hunk 4]
```diff
@@ -381,6 +384,17 @@ static TEE_Result stm32_rproc_probe(const void *fdt, int node,
 			goto err;
 	}
 
+#ifdef CFG_STM32MP15
+	if (!stm32_rcc_is_secure()) {
+		if (IS_ENABLED(CFG_INSECURE))
+			IMSG("WARNING: insecure rproc support regarding RCC hardening");
+		else
+			panic("RCC secure hardening issue");
```
Comment (GseoC) [inline core/drivers/remoteproc/stm32_remoteproc.c:392]: Can keep the trace above and replace WARNING: with ERROR:

File: core/arch/arm/plat-stm32mp1/shared_resources.c [hunk 5]
```diff
@@ -191,52 +191,6 @@ static void register_periph(enum stm32mp_shres id, enum shres_state state)
 	}
 
 	shres_state[id] = state;
-
-	/* Explore clock tree to lock secure clock dependencies */
```
Comment (GseoC) [inline core/arch/arm/plat-stm32mp1/shared_resources.c:195]: This should probably be part of another P-R, no?

**Comment by GseoC** [comment]
Commits:
`[plat-stm32mp1: shared_resources: don't check clock dependency](https://github.com/OP-TEE/optee_os/pull/7108/commits/8c281c6c79d98554953d7f2525469f2b26ab2445)` and `[plat-stm32mp1: remove stm32mp_register_clock_parents_secure()](https://github.com/OP-TEE/optee_os/pull/7108/commits/354742e829a3672ad7e0f1cd1777d4b46d308061) ` probably should be part of another P-R, no?

File: core/drivers/firewall/stm32_etzpc.c (hunk 3 above)
Comment (etienne-lms) [reply core/drivers/firewall/stm32_etzpc.c:180]: I think tt does: MCU isolation requires`stm32_rcc_is_secure() == true && stm32_rcc_is_mckprot() == true`

File: core/drivers/remoteproc/stm32_remoteproc.c (hunk 4 above)
Comment (etienne-lms) [reply core/drivers/remoteproc/stm32_remoteproc.c:392]: ok

File: core/arch/arm/plat-stm32mp1/shared_resources.c (hunk 5 above)
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/shared_resources.c:195]: This change could be squashed in next commit "plat-stm32mp1: remove stm32mp_register_clock_parents_secure()". Would you prefer this way?

**Review by etienne-lms — COMMENTED** [review]
Comments addressed.

File: core/arch/arm/plat-stm32mp1/shared_resources.c [hunk 6]
```diff
@@ -186,6 +186,9 @@ static void register_periph(enum stm32mp_shres id, enum shres_state state)
 			panic();
 		}
 		break;
+	case STM32MP1_SHRES_PLL3:
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/shared_resources.c:189]: There are still drivers calling `stm32mp_register_[non_]secure_xxx()` API functions. The addition here ensures none of them are registering PLL3 resource.
Once all drivers are changed to remove calls to`stm32mp_register_[non_]secure_xxx()`, we will be able to remove **plat-stm32mp1/shared_resources.c**.

[the remark this answers is not in the document]
Comment (GseoC) [reply core/arch/arm/plat-stm32mp1/shared_resources.c:189]: ok

**Comment by GseoC** [comment]
Can you please squash the commits?

**Comment by GseoC** [comment]
Nit: for consistency with other commits on ETZPC driver, maybe replace `firewall: stm32_etzpc` with `drivers: stm32_etzpc`

File: core/drivers/firewall/stm32_etzpc.c [hunk 7]
```diff
@@ -153,6 +156,44 @@ static enum etzpc_decprot_attributes etzpc_binding2decprot(uint32_t mode)
 	}
 }
 
+static void
+sanitize_decprot_config(uint32_t decprot_id __maybe_unused,
+			enum etzpc_decprot_attributes attr __maybe_unused)
```
Comment (GseoC) [inline core/drivers/firewall/stm32_etzpc.c:161]: attr is always used in the switch, please remove __maybe_unused

File: core/drivers/remoteproc/stm32_remoteproc.c [hunk 8]
```diff
@@ -381,6 +384,16 @@ static TEE_Result stm32_rproc_probe(const void *fdt, int node,
 			goto err;
 	}
 
+#ifdef CFG_STM32MP15
+	if (!stm32_rcc_is_secure()) {
+		IMSG("WARNING: insecure rproc isolation: RCC tzen:0");
```
Comment (GseoC) [inline core/drivers/remoteproc/stm32_remoteproc.c:389]: a bit too much of ":", proposition:
`WARNING: insecure rproc isolation, RCC tzen:0`

File: core/drivers/firewall/stm32_etzpc.c (hunk 7 above)
Comment (etienne-lms) [reply core/drivers/firewall/stm32_etzpc.c:161]: When `CFG_STM32MP15` is disabled, the function is empty.

Comment (GseoC) [reply core/drivers/firewall/stm32_etzpc.c:161]: Absolutely, my bad

**Comment by etienne-lms** [comment]
Comments addressed. Can I squash the fixup commit and fix the commit messages?

**Comment by GseoC** [comment]
Sure, again, maybe @arnopo would like to make a last review as well

File: core/drivers/remoteproc/stm32_remoteproc.c [hunk 9]
```diff
@@ -381,6 +384,16 @@ static TEE_Result stm32_rproc_probe(const void *fdt, int node,
...
+		if (IS_ENABLED(CFG_INSECURE))
```
Comment (arnopo) [inline core/drivers/remoteproc/stm32_remoteproc.c:390]:
```diff
-		if (IS_ENABLED(CFG_INSECURE))
+		if (!IS_ENABLED(CFG_INSECURE))
```

**Review by arnopo — CHANGES_REQUESTED** [review]

File: core/drivers/remoteproc/stm32_remoteproc.c (hunk 9 above)
Comment (etienne-lms) [reply core/drivers/remoteproc/stm32_remoteproc.c:390]: Right! thanks.

File: core/drivers/remoteproc/stm32_remoteproc.c [hunk 10]
```diff
@@ -381,6 +384,16 @@ static TEE_Result stm32_rproc_probe(const void *fdt, int node,
 			goto err;
 	}
 
+#ifdef CFG_STM32MP15
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/remoteproc/stm32_remoteproc.c:387]: Ok, i'll remove the condition. We'll bring that back when we'll add MP25 support.

**Comment by GseoC** [comment]
includes are not well sorted in stm32_etzpc.c (combination of my bad and one of you patch :)), with that addressed and all squashed:

`Reviewed-by: Gatien Chevallier <gatien.chevallier@foss.st.com>`

**Review by arnopo — APPROVED** [review]
Reviewed-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>

**Comment by etienne-lms** [comment]
Review tags applied.
CI / Code style reports are all false positive (trace message implementation exceeding 80char/line).

**Pull request merged**